File: javaSample/src/main/java/com/cleveradssolutions/sampleapp/rewarded/RewardedAdActivity.java
package com.cleveradssolutions.sampleapp.rewarded;

import static com.cleveradssolutions.sampleapp.SampleApplication.CAS_ID;

import android.os.Bundle;

import androidx.annotation.NonNull;

import com.cleveradssolutions.sampleapp.R;
import com.cleveradssolutions.sampleapp.util.BackNavigationActivity;
import com.cleveradssolutions.sampleapp.util.Util;
import com.cleveradssolutions.sdk.AdContentInfo;
import com.cleveradssolutions.sdk.AdFormat;
import com.cleveradssolutions.sdk.screen.CASRewarded;
import com.cleveradssolutions.sdk.screen.ScreenAdContentCallback;
import com.cleversolutions.ads.AdError;

public class RewardedAdActivity extends BackNavigationActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.screen_ad_activity);
        setTitle(R.string.adFormats_rewardedAd);

        CASRewarded rewardedAd = new CASRewarded(CAS_ID);
        rewardedAd.setAutoloadEnabled(false);

        rewardedAd.setContentCallback(new ScreenAdContentCallback() {
            @Override
            public void onAdLoaded(@NonNull AdContentInfo ad) {
                Util.toast(RewardedAdActivity.this, "Rewarded Ad loaded");
            }

            @Override
            public void onAdFailedToLoad(@NonNull AdFormat format, @NonNull AdError error) {
                Util.toastError(RewardedAdActivity.this, "Rewarded Ad failed to load: " + error.getMessage());
            }

            @Override
            public void onAdFailedToShow(@NonNull AdFormat format, @NonNull AdError error) {
                Util.toastError(RewardedAdActivity.this, "Rewarded Ad show failed: " + error.getMessage());
            }

            @Override
            public void onAdShowed(@NonNull AdContentInfo ad) {
                Util.toast(RewardedAdActivity.this, "Rewarded Ad shown");
            }

            @Override
            public void onAdClicked(@NonNull AdContentInfo ad) {
                Util.toast(RewardedAdActivity.this, "Rewarded Ad clicked");
            }

            @Override
            public void onAdDismissed(@NonNull AdContentInfo ad) {
                Util.toast(RewardedAdActivity.this, "Rewarded Ad closed");
            }
        });

        rewardedAd.setOnImpressionListener(ad ->
                Util.toast(this, "Rewarded Ad impression: " + ad.getRevenue())
        );

        findViewById(R.id.load).setOnClickListener(v -> rewardedAd.load(this));
        findViewById(R.id.show).setOnClickListener(v -> rewardedAd.show(this,
                (ad) -> Util.toast(this, "Reward earned"))
        );
    }

}

File: kotlinSample/src/main/java/com/cleveradssolutions/sampleapp/rewarded/RewardedAdActivity.kt
package com.cleveradssolutions.sampleapp.rewarded

import android.os.Bundle
import android.widget.Button
import com.cleveradssolutions.sampleapp.R
import com.cleveradssolutions.sampleapp.SampleApplication.Companion.CAS_ID
import com.cleveradssolutions.sampleapp.support.BackNavigationActivity
import com.cleveradssolutions.sampleapp.support.toast
import com.cleveradssolutions.sampleapp.support.toastError
import com.cleveradssolutions.sdk.AdContentInfo
import com.cleveradssolutions.sdk.AdFormat
import com.cleveradssolutions.sdk.OnAdImpressionListener
import com.cleveradssolutions.sdk.screen.CASRewarded
import com.cleveradssolutions.sdk.screen.ScreenAdContentCallback
import com.cleversolutions.ads.AdError

class RewardedAdActivity : BackNavigationActivity() {

    val screenContentCallback = object : ScreenAdContentCallback() {
        override fun onAdLoaded(ad: AdContentInfo) {
            toast("Rewarded Ad loaded")
        }

        override fun onAdFailedToLoad(format: AdFormat, error: AdError) {
            toastError("Rewarded Ad failed to load: ${error.message}")
        }

        override fun onAdFailedToShow(format: AdFormat, error: AdError) {
            toastError("Rewarded Ad show failed: ${error.message}")
        }

        override fun onAdShowed(ad: AdContentInfo) {
            toast("Rewarded Ad shown")
        }

        override fun onAdClicked(ad: AdContentInfo) {
            toast("Rewarded Ad clicked")
        }

        override fun onAdDismissed(ad: AdContentInfo) {
            toast("Rewarded Ad closed")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.screen_ad_activity)
        setTitle(R.string.adFormats_rewardedAd)

        val rewardedAd = CASRewarded(CAS_ID)
        rewardedAd.contentCallback = screenContentCallback
        rewardedAd.isAutoloadEnabled = false // by default

        // Optional set Impression listener
        rewardedAd.onImpressionListener = OnAdImpressionListener { ad ->
            toast("Rewarded Ad impression: ${ad.revenue}")
        }

        findViewById<Button>(R.id.load).setOnClickListener {
            rewardedAd.load(this)
        }

        findViewById<Button>(R.id.show).setOnClickListener {
            rewardedAd.show(this) {
                toast("Reward earned")
            }
        }
    }

}